src/Client.ts

```diff
@@ -667,7 +667,12 @@ export async function enterPassiveModeIPv6(client: Client): Promise<FTPResponse>
  */
 function parseIPv6PasvResponse(message: string): number {
     // Get port from EPSV response, e.g. "229 Entering Extended Passive Mode (|||6446|)"
-    const groups = message.match(/\|{3}(.+)\|/)
+    let groups = message.match(/\|{3}(.+)\|/)
```

Comment:
I'd prefer using a single regular expression, like e.g. `[|!]{3}(.+)[|!]` should be fine. Keep the comment about IBM, though.

Reply:
@patrickjuchli

I thought about going that route because I don't like the identical if check but I wasn't sure if I like the fact that that regexp could also match

!!!3843| or |||3843!

or am I seeing it wrong?

Reply:
You're seeing it correctly but we're writing a client, not a test for FTP servers and whether they adhere to a specification. We don't want t fail as quickly as possible. Is matching`!!!3843|` a problem? I don't see one.

Reply:
You are absolutely right! Will change the PR :)